@here/harp-mapview/lib/ClipPlanesEvaluator.ts

```diff
@@ -139,7 +215,7 @@ export class InterpolatedClipPlanesEvaluator implements ClipPlanesEvaluator {
  * ground surface (top view) along surface normal, but does not preserve correct planes when using
  * modifying camera pitch (tilt) angle. Suitable only top view camera modes.
  */
-export class AltitudeBasedClipPlanesEvaluator implements ClipPlanesEvaluator {
+export class AltitudeBasedClipPlanesEvaluator extends ElevationBasedClipPlanesEvaluator {
```

Comment:
Why do we actually need these two? What is the difference?

Reply:
I have provided the abstract class that allows for implementing other algorithms (classes) that still supports elevation, but maybe using different methods - simpler or more sophisticated. Later on we have two classes derived:
AltitudeBasedClipPlanesEvaluator - top view only (I would like it to rename it for that one, but not in the same patchset cause refactoring could make it even bigger),
TiltBasedClipPlanesEvaluator - full support for any camera angles.

Reply:
If I understand, you want a top view only far plane evaluator that supports evaluation. I don't think it makes sense to have 3D terrain if the user can't tilt.

Reply:
What's the difference between the altitudeBased and tiltBased whenever there's no tilt? Because results look really similar.

If they're indeed the same, and the altitudeBased can only be used in 2D mode, why not just using the tiltBased always?

Reply:
Basically only computation cost keeps us from using only TiltBased evaluator. If you know that the camera does not allow for tilt (as it is in current globe implementation), then AltitudeBased wins a lot. In the next commit I will rename them both to:
- TopViewClipPlanesEvaluator,
- TiltViewClipPlanesEvaluator.
just to clarify the things. The names are quite old and comes from the formulas used, by the time both evaluators changed a lot so no need to keep the old naming.